server/http/restapi/marketplace.go

```diff
@@ -0,0 +1,145 @@
+package restapi
+
+import (
+	"net/http"
+	"sort"
+	"strings"
+
+	"github.com/mattermost/mattermost-plugin-apps/apps"
+	"github.com/mattermost/mattermost-plugin-apps/server/utils/httputils"
+
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/pkg/errors"
+)
+
+type MarketplaceApp struct {
+	Installed bool                     `json:"installed"`
+	Labels    []model.MarketplaceLabel `json:"labels,omitempty"`
+	Manifest  apps.Manifest            `json:"manifest"`
+}
+
+func (a *restapi) handleGetMarketplace(w http.ResponseWriter, req *http.Request, actingUserID string) {
+	filter := req.URL.Query().Get("filter")
+
+	localApps, err := a.getLocalApps()
+	if err != nil {
+		httputils.WriteInternalServerError(w, err)
+		return
+	}
+
+	remoteApps, err := a.getRemoteApps()
+	if err != nil {
+		httputils.WriteInternalServerError(w, err)
+		return
+	}
+
+	apps := mergeApps(remoteApps, localApps)
+
+	// Filter plugins.
+	var result []MarketplaceApp
+	for _, a := range apps {
+		if appMatchesFilter(a.Manifest, filter) {
+			result = append(result, a)
+		}
+	}
+
+	// Sort result alphabetically.
+	sort.SliceStable(result, func(i, j int) bool {
+		return strings.ToLower(result[i].Manifest.DisplayName) <
+			strings.ToLower(result[j].Manifest.DisplayName)
+	})
+
+	httputils.WriteJSON(w, result)
+}
+
+func (a *restapi) getLocalApps() ([]MarketplaceApp, error) {
+	apps, _, err := a.api.Admin.ListApps()
```

Comment:
Just to clarify, apps stored here are all apps `registered` for this installation. Some of them might be already installed by the instance admin, others might not. Installed apps might be `enabled` or `disabled`.

Reply:
Should `master` already behave that way? Right now Apps are only stored when they get installed or provisioned.

I'm hesitant to base the PR on changes that are not yet in `master`.

Reply:
No. [PR](https://github.com/mattermost/mattermost-plugin-apps/pull/46) is not merged yet.